q2_diversity/_alpha.py

```diff
@@ -5,3 +5,110 @@
+
+import os
+
+import biom
+import scipy
+import skbio
+import qiime
+import pandas as pd
+import seaborn as sns
+import skbio.diversity
+from statsmodels.sandbox.stats.multicomp import multipletests
+
+
+# We should consider moving these functions to scikit-bio. They're part of
+# the private API here for now.
+def phylogenetic_metrics():
+    return {'faith_pd'}
+
+
+def non_phylogenetic_metrics():
+    return {'ace', 'chao1', 'chao1_ci', 'berger_parker_d', 'brillouin_d',
+            'dominance', 'doubles', 'enspie', 'esty_ci', 'fisher_alpha',
+            'goods_coverage', 'heip_e', 'kempton_taylor_q', 'margalef',
+            'mcintosh_d', 'mcintosh_e', 'menhinick', 'michaelis_menten_fit',
+            'observed_otus', 'osd', 'pielou_e', 'robbins', 'shannon',
+            'simpson', 'simpson_e', 'singles', 'strong', 'gini_index',
+            'lladser_pe', 'lladser_ci'}
+
+
+def alpha_phylogenetic(metric: str, feature_table: biom.Table,
+                       phylogeny: skbio.TreeNode=None) -> pd.Series:
+    if metric not in phylogenetic_metrics():
+        raise ValueError("Unknown phylogenetic metric: %s" % metric)
+
+    counts = feature_table.matrix_data.toarray().astype(int).T
+    sample_ids = feature_table.ids(axis='sample')
+    feature_ids = feature_table.ids(axis='observation')
+
+    return skbio.diversity.alpha_diversity(
+              metric=metric,
+              counts=counts,
+              ids=sample_ids,
+              otu_ids=feature_ids,
+              tree=phylogeny)
+
+
+def alpha(metric: str, feature_table: biom.Table) -> pd.Series:
+    if metric not in non_phylogenetic_metrics():
+        raise ValueError("Unknown metric: %s" % metric)
+
+    counts = feature_table.matrix_data.toarray().astype(int).T
+    sample_ids = feature_table.ids(axis='sample')
+
+    return skbio.diversity.alpha_diversity(
+              metric=metric,
+              counts=counts,
+              ids=sample_ids)
+
+
+def alpha_comparisons(output_dir: str, alpha_diversities: pd.Series,
+                      metadata: qiime.MetadataCategory) -> None:
+    alpha_diversities.name = 'Alpha diversity'
+    metadata = metadata.to_series()
+    data = pd.concat([alpha_diversities, metadata], axis=1)
+    names = []
+    groups = []
+    for name, group in data.groupby(metadata.name):
+        names.append(name)
+        groups.append(list(group['Alpha diversity']))
+    ax = sns.boxplot(data=groups)
+    ax.set_xticklabels(names)
+    ax.get_figure().savefig(os.path.join(output_dir, 'boxplots.png'))
+    ax.get_figure().savefig(os.path.join(output_dir, 'boxplots.pdf'))
+
+    # perform Kruskal-Wallis H-test
+    H, p = scipy.stats.mstats.kruskalwallis(*groups)
+
+    # perform Mann-Whitney U-test and correct for multiple comparisons
+    u_results = []
+    for i in range(len(names)):
+        for j in range(i):
+            U, p = scipy.stats.mannwhitneyu(groups[i], groups[j])
```

Comment:
beware of this function, it's had a slew of bugs and backwards-incompatible API changes (some of which aren't documented). we ended up dropping it's use from skbio for those reasons. not sure how necessary this function is, or if there's a similar method available to easily drop in